#! /bin/sh -e
## 05_hppabuild.dpatch by Junichi Uekawa <dancer@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: patch to get HPPA Linux build. applied upstream on 3.7p1

if [ $# -ne 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi
case "$1" in
    -patch) patch -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

--- canna-3.7.orig/Canna.conf.dist
+++ canna-3.7/Canna.conf.dist
@@ -183,7 +183,7 @@
 #if defined(cygwinArchitecture)
 /* import libraryの方を指定しておく。DLL本体は/usr/X11R6/binに入る。 */
 sharedLibExtension = a
-#elif defined(HPArchitecture)
+#elif defined(HPArchitecture) && !defined(__linux__)
 sharedLibExtension = sl.$(cannaDsoRev)
 #elif defined(DarwinArchitecture)
 sharedLibExtension = $(cannaDsoRev).dylib
